### initial values for transformation estimates
# insert the values as x, y, z, yaw (rot around z), pitch (rot around y), roll (rot around x)
# the transform from base_frame to torso_lower_frame
T_base_to_torso_lower_initial: [0.3, 0, 0.69, 0.0, 0.0, 0.0]

# the transform from torso_upper_frame to camera_frame
T_torso_upper_to_camera_initial: [0.015, 0.065, 0.0, 0.0, 0.0, -1.57]


### Observation positions for capturing calibration images of the pitags
# if this flag is true, then the robot configurations will be sampled on a equally spaced grid from the given ranges, if false, the user-provided configurations in robot_configurations will be used
# bool
use_range: true

# ranges for the degrees of freedom of the robot
# each parameter is set as [min_value, step, max_value]
# double
x_range: [-1.5, 0.25, -0.75]       # in [m]
y_range: [-0.75, 0.25, -1.5]       # in [m]
phi_range: [0.0, 1.0, 0.0]         # in [rad], fixed phi-angle
pan_range: [-0.7, 0.1, 1.5]     # in [rad]
tilt_range: [-0.5, 0.1, 1.5]    # in [rad]
#pan_range: [-0.52, 0.02, 0.52]     # in [rad]
#tilt_range: [-0.44, 0.02, 0.44]    # in [rad]

# list of robot configurations for observing the pitags measured relative to the landmark_reference_nav coordinate system that is similarly aligned as the robot's base_link facing the marker
# includes 5 parameters per entry: robot pose: x, y, phi and torso: pan, tilt
robot_configurations: [-1.5, -0.17, 0, 0.15, 0.25,
                       -1.5, -0.17, 0, 0.0, 0.3,
                       -1.5, -0.17, 0, -0.15, 0.3,
                       -1.5, -0.17, 0, -0.3, 0.3,
                       -1.5, -0.17, 0, -0.5, 0.3,
                       -1.5, -0.17, 0, 0.15, 0.05,
                       -1.5, -0.17, 0, 0.0, 0.05,
                       -1.5, -0.17, 0, -0.15, 0.05,
                       -1.5, -0.17, 0, -0.3, 0.05,
                       -1.5, -0.17, 0, -0.5, 0.05,
                       -1.5, -0.17, 0, 0.15, -0.2,
                       -1.5, -0.17, 0, 0.0, -0.2,
                       -1.5, -0.17, 0, -0.15, -0.2,
                       -1.5, -0.17, 0, -0.35, -0.2,
                       -1.5, -0.17, 0, -0.5, -0.2,
                       -1.0, -0.17, 0, 0.0, 0.2,
                       -1.0, -0.17, 0, -0.2, 0.2,
                       -1.0, -0.17, 0, -0.45, 0.2,
                       -1.0, -0.17, 0, 0.0, 0.05,
                       -1.0, -0.17, 0, -0.2, 0.05,
                       -1.0, -0.17, 0, -0.45, 0.05,
                       -1.0, -0.17, 0, 0.0, -0.15,
                       -1.0, -0.17, 0, -0.2, -0.15,
                       -1.0, -0.17, 0, -0.45, -0.15,
                       -0.85, -0.17, 0, 0.0, 0.15,
                       -0.85, -0.17, 0, -0.15, 0.15,
                       -0.85, -0.17, 0, -0.35, 0.2,
                       -0.85, -0.17, 0, 0.0, 0.05,
                       -0.85, -0.17, 0, -0.15, 0.05,
                       -0.85, -0.17, 0, -0.35, 0.05,
                       -0.85, -0.17, 0, 0.0, -0.1,
                       -0.85, -0.17, 0, -0.15, -0.1,
                       -0.85, -0.17, 0, -0.35, -0.1]


# link names for the robot coordinate systems
# string
# fixed frame at lower torso end (i.e. this coordinate system should not turn if different pan angles are commanded)
torso_lower_frame: "base_pan_link"

# last link of torso chain [the transformations between torso_lower_frame and torso_upper_frame should be available from tf]
torso_upper_frame: "tilt_link"

# camera frame with fixed transform to torso_upper_frame [transform from torso_upper_frame to camera will be calibrated by this program]
camera_frame: "kinect_link"

# this is the camera coordinate system which refers to the color image sensor [the transformations between camera_frame and camera_optical_frame should be available from tf]
camera_optical_frame: "kinect_rgb_optical_frame"

# the robot base frame [the transformation between laser scanner and base should be accomplished before, the transform from base_frame to torso_lower_framewill be calibrated by this program]
base_frame: "base_link"

# base name of all frames generated by cob_fiducials, each Pi tag number will be attached to this string.
marker_frame_base_name: "tag"


### program sequence
# loads calibration images and transforms from disk if set to true
# bool
load_images: false

# number of optimization iterations
# int
optimization_iterations: 1000
